x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/registry/ModelRegistry.java

```diff
@@ -917,43 +917,135 @@ private static Model getModelFromMap(@Nullable String inferenceId, Map<String, M
         return null;
     }
 
-    private void updateClusterState(List<Model> models, ActionListener<AcknowledgedResponse> listener, TimeValue timeout) {
-        var inferenceIdsSet = models.stream().map(Model::getInferenceEntityId).collect(Collectors.toSet());
-        var storeListener = listener.delegateResponse((delegate, exc) -> {
-            logger.warn(format("Failed to add minimal service settings to cluster state for inference endpoints %s", inferenceIdsSet), exc);
-            deleteModels(
-                inferenceIdsSet,
-                ActionListener.running(
-                    () -> delegate.onFailure(
-                        new ElasticsearchStatusException(
-                            format(
-                                "Failed to add the inference endpoints %s. The service may be in an "
-                                    + "inconsistent state. Please try deleting and re-adding the endpoints.",
-                                inferenceIdsSet
-                            ),
-                            RestStatus.INTERNAL_SERVER_ERROR,
-                            exc
+    private void updateClusterState(ResponseInfo responseInfo, ActionListener<AcknowledgedResponse> listener, TimeValue timeout) {
+        var inferenceIdsSet = responseInfo.successfullyStoredModels().stream().map(Model::getInferenceEntityId).collect(Collectors.toSet());
+
+        SubscribableListener.<Void>newForked(outOfSyncListener -> handleOutOfSyncEndpoints(responseInfo, outOfSyncListener, timeout))
+            .<AcknowledgedResponse>andThen(addModelMetadataTaskListener -> {
+                var cleanupListener = addModelMetadataTaskListener.delegateResponse((delegate, exc) -> {
+                    logger.atWarn()
+                        .withThrowable(exc)
+                        .log("Failed to add minimal service settings to cluster state for inference endpoints {}", inferenceIdsSet);
```

Comment:
```suggestion
                        .log("Failed to add minimal service settings to cluster state for inference endpoints [{}]", inferenceIdsSet);
```

Not sure, if log messages also usually use the `[element1, ..., elementN]` notation. Feel free to ignore